core/src/components/datetime/datetime.tsx

```diff
@@ -995,9 +997,12 @@ export class Datetime implements ComponentInterface {
     const nextMonth = calendarBodyRef.querySelector('.calendar-month:last-of-type');
     if (!nextMonth) { return; }
 
+    const isRTL = document.dir === 'rtl';
```

Comment:
We need to check if `dir="rtl"` was set on the host element as well. Might be good to make this a utility that other components can use as well. Doing a quick search reveals that other components do not check the host when they probably should: https://github.com/ionic-team/ionic-framework/search?l=TSX&p=2&q=rtl

Reply:
Added a new util for `isRTL`. 

How do you feel about splitting out a separate issue for each component that has rtl checks only for document? That way the change log will better reflect those changes to the util change and won't be tied to the datetime changes. If you feel good about that, I'm happy to create an issue for each.

Reply:
Yep let's make some tech debt for that 👍